### PR TITLE
Extract component for moving files to rejected container

### PR DESCRIPTION
**Pull request by rkondratowicz**
The new job that handles duplicate files will need to move duplicates to rejected container.

Extracting code doing that from already existing component.

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandlerTest.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
     @BeforeEach
     void setUp() {
         dbHelper.deleteAll();
-        mover = new RejectedFilesHandler(storageClient, envelopeService);
+        mover = new RejectedFilesHandler(envelopeService, new BlobRejecter(storageClient));
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandlerTest.java]: This class no longer accesses blob storage directly.
I'll remove integration test for it and introduce a new one (for `BlobRejecter`) in the next PR.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    public void moveToRejectedContainer(String blobName, String containerName) {
+
+        var sourceBlob = storageClient.getBlobContainerClient(containerName).getBlobClient(blobName);
+        var targetBlob = storageClient.getBlobContainerClient(containerName + REJECTED_CONTAINER_SUFFIX).getBlobClient(
+            blobName);
+
+        if (targetBlob.exists()) {
+            targetBlob.createSnapshot();
+        }
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: perhaps we should create snapshot right before copying the blob?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: Good point!
Just moving existing code to a new class in this PR.
I'll fix this in a separate PR (and add a test - no need to create a snapshot if file in source container doesn't exist).

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        String loggingContext = String.format(
+            "File name: %s. Source Container: %s. Target Container: %s",
+            blobName,
+            sourceBlob.getContainerName(),
+            targetBlob.getContainerName()
+        );
+
+        if (!sourceBlob.exists()) {
+            logger.error("File already deleted. " + loggingContext);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: Maybe worth saying a bit more, i.e. that the file was supposed to be moved to rejected container.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        } else {
+            copy(targetBlob, sourceBlob, loggingContext);
+            sourceBlob.delete();
+            logger.info("File successfully moved to rejected container. " + loggingContext);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: After `copy` we know that the file has been copied. Maybe in this last `info` we should say that the source file was deleted.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.blobrouter.services.storage.RejectedFilesHandler.REJECTED_CONTAINER_SUFFIX;
+
+@Component
+public class BlobRejecter {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: The blob is rejected earlier and this tool just moves it. Maybe `RejectedBlobMover` or something like that?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: Note that the duplicate handler will use this class as well

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: moved existing code

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java (hunk 4 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: moved existing code

**Comment by lgonczar** [comment]
Nice idea. It would be good to have separate tests for this new class, even if they are integration tests.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java (hunk 5 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: So `BlobMover`?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java]: renamed to `BlobMover`, method is called `moveToRejectedContainer`

**Pull request merged**